packages/react-core/src/components/MultipleFileUpload/examples/MultipleFileUploadBasic.tsx

```diff
@@ -0,0 +1,150 @@
+
+    if (readFileData.every(file => file.loadResult === 'success')) {
+      return <CheckCircleIcon />;
+    }
+
+    return <TimesCircleIcon />;
+  };
+
+  // remove files from both state arrays based on their name
+  const removeFiles = (namesOfFilesToRemove: string[]) => {
+    const newCurrentFiles = currentFiles.filter(
+      currentFile => !namesOfFilesToRemove.some(fileName => fileName === currentFile.name)
+    );
+
+    setCurrentFiles(newCurrentFiles);
+
+    const newReadFiles = readFileData.filter(
+      readFile => !namesOfFilesToRemove.some(fileName => fileName === readFile.fileName)
+    );
+
+    setReadFileData(newReadFiles);
+  };
+
+  // callback that will be called by the react dropzone with the newly dropped file objects
+  const handleFileDrop = (droppedFiles: File[]) => {
+    // identify what, if any, files are re-uploads of already uploaded files
+    const currentFileNames = currentFiles.map(file => file.name);
+    const reUploads = droppedFiles.filter(droppedFile => currentFileNames.includes(droppedFile.name));
+
+    /** this promise chain is needed because if the file removal is done at the same time as the file adding react
+     * won't realize that the status items for the re-uploaded files needs to be re-rendered */
+    Promise.resolve()
+      .then(() => removeFiles(reUploads.map(file => file.name)))
+      .then(() => setCurrentFiles(prevFiles => [...prevFiles, ...droppedFiles]));
+  };
+
+  // callback called by the status item when a file is successfully read with the built-in file reader
+  const handleReadSuccess = (data: string, file: File) => {
+    setReadFileData(prevReadFiles => [...prevReadFiles, { data, fileName: file.name, loadResult: 'success' }]);
+  };
+
+  // callback called by the status item when a file encounters an error while being read with the built-in file reader
+  const handleReadFail = (error: DOMException, file: File) => {
+    setReadFileData(prevReadFiles => [
+      ...prevReadFiles,
+      { loadError: error, fileName: file.name, loadResult: 'danger' }
+    ]);
+  };
+
+  // dropzone prop that communicates to the user that files they've attempted to upload are not an appropriate type
+  const handleDropRejected = (files: File[], _event: React.DragEvent<HTMLElement>) => {
+    if (files.length === 1) {
+      setModalText(`${files[0].name} is not an accepted file type`);
+    } else {
+      const rejectedMessages = files.reduce((acc, file) => (acc += `${file.name}, `), '');
+      setModalText(`${rejectedMessages}are not accepted file types`);
+    }
+  };
+
+  const successfullyReadFileCount = readFileData.filter(fileData => fileData.loadResult === 'success').length;
+
+  return (
+    <MultipleFileUpload
+      onFileDrop={handleFileDrop}
+      dropzoneProps={{
+        accept: 'image/jpeg, application/msword, application/pdf, image/png',
+        onDropRejected: handleDropRejected
+      }}
+    >
+      <MultipleFileUploadMain>
+        <MultipleFileUploadTitle>
```

Comment:
@mcoker Are the icon and titles flexible (they can go in any order)?  If not I would add a prop rather allowing consumer to add it themselves.

Reply:
@tlabaj technically they can, but each is assigned to a specific place in the grid currently - so while you can re-order them in the code, the order will not change visually. I suppose the only thing you could do by re-ordering them is to change where they are in the DOM for assistive tech or tab order reasons.

Reply:
@tlabaj technically they can since they're part of a grid with a gap, so you could put the icon after the title if you wanted to, but that does go against the design and I suspect we don't want to allow that. @mcarrano @mmenestr wdyt?

Reply:
Agree. I wouldn't want those to be reordered.